### PR TITLE
Add checks for Subresource identity and an EmptyReturn type

### PR DESCRIPTION
**Pull request by hansl**
The EmptyReturn is to simplify the return types of module methods.

### CODE REVIEW

File: src/many/src/server/module/_7_kvstore_commands/delete.rs [hunk 1]
```diff
@@ -8,6 +9,4 @@ pub struct DeleteArgs {
     pub key: ByteVec,
 }
 
-#[derive(Encode, Decode)]
-#[cbor(map)]
-pub struct DeleteReturns {}
+pub type DeleteReturn = EmptyReturn;
```
Comment (fmorency) [inline src/many/src/server/module/_7_kvstore_commands/delete.rs:12]: Not sure I understand this one. Care to shed some light?

The `delete` return value in the spec is `{}`. If I understand correctly, this change would imply that the return value of `delete` is an empty map?

**Review by fmorency — CHANGES_REQUESTED** [review]
Looking good! A few minor comments/questions.

Thanks!

File: src/many/src/protocol.rs [hunk 2]
```diff
@@ -24,18 +24,19 @@ impl Attribute {
         }
     }
 
-    pub const fn new(id: AttributeId, arguments: Vec<CborAny>) -> Self {
+    pub fn new(id: AttributeId, arguments: Vec<CborAny>) -> Self {
```
[the remark this answers is not in the document]
Comment (hansl) [reply src/many/src/protocol.rs:27]: No. This was from before we talked.

File: src/many/src/server/module/_7_kvstore_commands/delete.rs (hunk 1 above)
Comment (hansl) [reply src/many/src/server/module/_7_kvstore_commands/delete.rs:12]: Delete is NOT `{}`, it's `()` which is anything.

Comment (hansl) [reply src/many/src/server/module/_7_kvstore_commands/delete.rs:12]: We encode it as a value so it doesn't break decoders that would expect one, but it shouldn't have any value, really. Empty returns are 0-byte semantically, but we don't want to break CBOR.

**Review by fmorency — APPROVED** [review]

**Pull request merged**